hs_core/tests/api/native/test_reorder_authors_management_command.py

```diff
@@ -0,0 +1,183 @@
+        call_command(self.update_command)
+
+        self.assertEqual(self.res.get_citation(), citation_original)
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+    def test_command_alters_if_incorrect(self):
+        """
+        Testing author order is updated if it was incorrect
+        """
+
+        # Intentionally make a creator list with duplicate orders
+        john = self.res.metadata.creators.filter(name="John Smith").first()
+        lisa = self.res.metadata.creators.filter(name="Lisa McWill").first()
+        self.assertEqual(john.order, 2)
+        john.order = 3
+        john.save()
+        self.assertEqual(john.order, lisa.order)
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+    def test_command_alters_missing_first_author(self):
+        """
+        Testing author order is updated if there was a missing first author
+        """
+
+        # Intentionally make a creator list with duplicate orders
+        john = self.res.metadata.creators.filter(name="John Smith").first()
+        first_author = self.res.metadata.creators.filter(order=1).first()
+        self.assertEqual(first_author.order, 1)
+        first_author.order = 2
+        first_author.save()
+        self.assertEqual(john.order, first_author.order)
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+        self.assertEqual(first_author.name, "Creator_LastName, Creator_FirstName")
+
+    def test_command_fixes_multiple_authors(self):
+        """
+        Testing author order is updated if there were multiple duplicate author orders
+        """
+
+        # Intentionally make a creator list with duplicate orders
+        lisa = self.res.metadata.creators.filter(name="Lisa McWill").first()
+        last_author = self.res.metadata.creators.last()
+        self.assertEqual(lisa.order, 3)
+        self.assertEqual(last_author.order, 5)
+        lisa.order = 2
+        last_author.order = 4
+        lisa.save()
+        last_author.save()
+
+        cit_original = self.res.get_citation()
+
+        # run  update command to fix author order
+        call_command(self.update_command)
+
+        for index, creator in enumerate(self.res.metadata.creators.all(), start=1):
+            self.assertEqual(index, creator.order)
+
+        self.assertEqual(self.res.get_citation(), cit_original)
+
+        hs_author = self.res.metadata.creators.filter(email="user1@nowhere.com").first()
+        mark = self.res.metadata.creators.filter(name="Mark Miller").first()
+        self.assertIn(hs_author.order, [1, 2])
```

Comment:
I am not following why we are checking order can be either 1 or 2.

Reply:
good point, thanks for catching this. It was left over and needed to be removed.